Added AC_BUILD_STEP_STATUS to reserved env vars (#786)
appcircleio/appcircle-docusaurus: docs/environment-variables/appcircle-specific-environment-variables.md

| AC_INTERNAL_CONFIGURATION_NAME | Name of the configuration that used the build.                                               |
| AC_INTERNAL_CONFIGURATION_ID | Unique identifier for the [configuration](https://docs.appcircle.io/build/build-process-management/build-profile-configuration) that started the build. |
| AC_PROVIDER_NAME         | **Git Provider**. Options include: GitHub, GitHub App, GitLab, GitLab Self-Hosted, Bitbucket, Bitbucket Server, Azure DevOps Services. |

> | AC_IS_SUCCESS            | Set to `true` if the previous step was successful. This environment variable will be **deprecated** soon. You can use `AC_BUILD_STATUS` instead. |

Reviewer: Deprecated now, will be removed.

```suggestion
| AC_IS_SUCCESS            | Set to `true` if the previous step was successful. This environment variable is **deprecated** and might be removed in the future. You can use `AC_BUILD_STATUS` instead. |
```

| AC_BUILD_STATUS          | Specify the status of the build that has run so far. Options: `Success`, `Warning`, and `Failed`. |

> | AC_BUILD_STEP_STATUS | Provides detailed information about the status of the build steps executed so far. Steps that are disabled will not appear in this environment variable. The JSON output for executed steps includes the following fields: <br />- **StepName**: The name of the executed step. <br />- **BuildStatus**: The status of the step. Possible values: `Success`, `Warning`, or `Failed`. <br />- **Duration**: The time taken to complete the step, represented in seconds (e.g., `0.0000000`). <br />- **StartDate**: The start time of the step, formatted as an ISO 8601 timestamp (e.g., `2024-12-13T15:45:59.6426984Z`). <br />- **FinishDate**: The completion time of the step, also formatted as an ISO 8601 timestamp (e.g., `2024-12-13T15:45:59.6426984Z`). <br /> For additional details and instructions on how to format this output for readability, refer to the [**How can I print the status of workflow steps with detailed information?**](/workflows/common-workflow-steps/custom-script#how-can-i-print-the-status-of-workflow-steps-with-detailed-information) documentation. |

Reviewer: Is the naming correct? In the sample script and screenshots from issue: `AC_BUILD_STEPS_STATUS`
